src/ws/cockpitauth.c

```diff
@@ -521,12 +521,14 @@ session_start_process (const gchar **argv,
 static void
 send_authorize_reply (CockpitTransport *transport,
                       const gchar *cookie,
-                      const gchar *authorization)
+                      const gchar *authorization,
+                      const gchar *remote_peer)
 {
   const gchar *fields[] = {
     "command", "authorize",
     "cookie", cookie,
     "response", authorization,
+    remote_peer ? "remote-peer" : NULL, remote_peer,
```

Comment:
It's not about ordering, that doesn't matter any more since commit 9709cb121f02. My previous version just always set the key, but I changed it to emphasize that the key is optional; but that doesn't really change behaviour. I don't really feel like reworking the whole thing now -- the whole ws is due for a bigger rewrite anyway, and it's mostly stylistic busywork IMHO.

I can also just set it to `remote_peer ? ""` as before if you like that better?

Reply:
I feel like it's more correct not to add it if we don't have it, so if your answer is "I don't feel like rewriting that" then let's just leave it?

One more thing you could do is to write the value `null`.  That's valid JSON, and the way our parser is written, I think it would fail to find any value not starting with a `"`, right?

Or just leave it as-is.

Reply:
The parser in c-session is fine with `null`. What I'm not sure about is if `node = json_node_init_string (json_node_new (JSON_NODE_VALUE), fields[i]);` will get along with `NULL`. As said, in practice it doesn't happen that remote-peer is ever unknown, so I'm nervous to have a NULL string rendered there without being able to test it.. So I'd prefer to leave it as-is, if that's ok with you?